Free Listings + Paid Ads: Add the product feed status section UI

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

This PR implements the UI part of **📌 Product feed status section** in #1610.

- Extract a code fragment from the sync-status.js as `getNumberOfSyncProducts` function.
- Implement `ProductFeedStatusSection` component and add it to the paid ads setup step of onboarding flow.

### Screenshots:

#### 📷 Normal view

![image](https://user-images.githubusercontent.com/17420811/185092659-2c9944e3-6c3e-4f32-b62b-896b3099b1cb.png)

#### 📷 Hover on the "# products" text

![image](https://user-images.githubusercontent.com/17420811/185092718-f3e423e8-767e-4a8f-b54a-b7547754fbe6.png)

### Detailed test instructions:

1. Check if the visual result of the "Product feed status" section is close to the design in Figma.
2. Hover on the "# products" text to view the tooltip.

### Changelog entry

eason9487 committed:
Extract a code fragment from the sync-status.js as `getNumberOfSyncProducts` function

eason9487 committed:
Implement ProductFeedStatusSection component and add it to the paid ads setup step of onboarding flow

ianlin commented on js/src/data/actions.js:
```diff
@@ -95,6 +95,24 @@ export function handleFetchError( error, message ) {
+/**
+ * Product status statistics on Google Merchant Center
+ *
+ * @typedef {Object} ProductStatistics
```
💅  Since the type of `ProductStatistics` is defined, it would be nice to add some doc comments in the below function that is using this type.

https://github.com/woocommerce/google-listings-and-ads/blob/9333fa5a0c2a69dc0aa177cd119be07bafcd3698/js/src/data/actions.js#L892

eason9487 replied:
Added in https://github.com/woocommerce/google-listings-and-ads/pull/1638/commits/456668361dccc992d835445f05ae3f24cfebd8fb.

ianlin commented on js/src/setup-mc/setup-stepper/setup-paid-ads/product-feed-status-section.js:
```diff
@@ -0,0 +1,113 @@
+ * and show the number of products will be synced to Google Merchant Center.
+ */
+export default function ProductFeedStatusSection() {
+	/*
+	const { data, hasFinishedResolution } = useAppSelectDispatch(
+		'getMCProductStatistics'
+	);
+	*/
+	// TODO: Replace the dummy data with the above code later to use the adjusted API.
```
❔ Just wanted to make sure my understanding is right. The API `mc/product-statistics` would only be returning data after we call `mc/settings/sync`, so the comment `Replace the dummy data with the above code later to use the adjusted API.` means we would make a POST request to  `mc/settings/sync` later, right?

eason9487 replied:
Yes, as per my test from the frontend side, it has to call `mc/settings/sync` before getting statistics from `mc/product-statistics`, so the frontend side will call `mc/settings/sync` when entering step 4. And the `mc/product-statistics` seems to check whether the `MC_SETUP_COMPLETED_AT` flag is set before returning, thus, we will need the related adjustment in this project as well.

ianlin reviewed (approved):
LGTM.

✅  The UI looks close to the design.
✅  The tooltip was displayed when hovering on the `# products`.

eason9487 committed:
Add JSDoc for more functions that also relate to the `ProductStatistics` type.

Address https://github.com/woocommerce/google-listings-and-ads/pull/1638#discussion_r948706947
